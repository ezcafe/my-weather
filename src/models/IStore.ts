import { RouterState } from 'connected-react-router';
import IErrorState from '../stores/error/models/IErrorState';
import ILocationsState from '../stores/locations/models/ILocationsState';
import IRequestingState from '../stores/requesting/models/IRequestingState';
import IToastsState from '../stores/toasts/models/IToastsState';

export default interface IStore {
    readonly error: IErrorState;
    readonly requesting: IRequestingState;
    readonly router: RouterState;
    readonly locations: ILocationsState;
    readonly toasts: IToastsState;
}
