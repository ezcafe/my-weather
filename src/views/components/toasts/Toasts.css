.toasts {
    position: absolute;
    top: 46px;
    right: 10px;
    width: 300px;
}
