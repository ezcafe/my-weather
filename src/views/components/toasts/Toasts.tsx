import { useSelector } from 'react-redux';
import { toastsSelector } from '../../../stores/toasts/ToastsSelector';
import IToast from '../../../stores/toasts/models/IToast';
import ToastCard from './Toast';
import './Toasts.css';

const Toasts: React.FC = () => {
    const toasts: IToast[] = useSelector(toastsSelector);

    if (toasts.length === 0) {
        return null;
    }

    return (
        <div className="toasts">
            {toasts.map((model: IToast) => (
                <ToastCard key={model.id} item={model} />
            ))}
        </div>
    );
};

export default Toasts;
