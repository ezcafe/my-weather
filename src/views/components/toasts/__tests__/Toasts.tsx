import { render, RenderResult } from '@testing-library/react';
import Toasts from '../Toasts';
import { mockedToasts } from '../../../../fixtures';
import withMockedRedux from '../../../../fixtures/withMockedRedux';
import IStore from '../../../../models/IStore';

describe('Toasts', () => {
    function renderComponent(initialState?: Partial<IStore>): RenderResult {
        const WrappedComponent = withMockedRedux(Toasts, initialState);
        return render(<WrappedComponent />);
    }

    test('should show component', () => {
        const { container } = renderComponent({
            toasts: mockedToasts,
        });
        expect(container).toMatchSnapshot();
    });
});
