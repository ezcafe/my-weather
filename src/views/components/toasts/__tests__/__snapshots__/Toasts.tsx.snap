// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toasts should show component 1`] = `
<div>
  <div
    class="toasts"
  >
    <div
      aria-atomic="true"
      aria-live="assertive"
      class="fade toast show"
      role="alert"
    >
      <div
        class="toast-header"
      >
        <strong
          class="mr-auto"
        >
          error
        </strong>
        <button
          class="close ml-2 mb-1"
          data-dismiss="toast"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
          <span
            class="sr-only"
          >
            Close
          </span>
        </button>
      </div>
      <div
        class="toast-body"
      >
        Error message
      </div>
    </div>
  </div>
</div>
`;

exports[`Toasts should show nothing 1`] = `<div />`;
